### PR TITLE
Tweak

### PR DESCRIPTION
**Pull request by saas786**
- Upgraded Hybrid Core usage
- adding purchased order items name as tags in ChartMogul

### CODE REVIEW

**Comment by pawelkmpt** [comment]
Hybrid Core is updated in https://github.com/saas786/cxl-common-lib/commit/fd63cce85a84fe35e72e881be9cfb5094728e2d6, right?

**Review by pawelkmpt — APPROVED** [review]

**Comment by saas786** [comment]
> Hybrid Core is updated in [saas786/cxl-common-lib@fd63cce](https://github.com/saas786/cxl-common-lib/commit/fd63cce85a84fe35e72e881be9cfb5094728e2d6), right?

Yes correct:
CXL Common Lib latest:
https://github.com/saas786/cxl-common-lib/releases/tag/v1.0.0-RC.1
https://github.com/saas786/cxl-common-lib/releases/download/v1.0.0-RC.1/cxl-common-lib.zip

**Comment by lkraav** [comment]
@pawelkmpt is this OK to merge on staging?

I am looking into things and would like the latest tech available as the baseline.

**Comment by saas786** [comment]
> @pawelkmpt is this OK to merge on staging?
> 
> I am looking into things and would like the latest tech available as the baseline.

Yes from me, its latest. Pawel can take a look.

**Comment by lkraav** [comment]
We don't need to self-host Sentry SDK, we're already using https://github.com/stayallive/wp-sentry

File: src/Export/Component.php [hunk 1]
```diff
@@ -202,19 +202,28 @@ private function createOrder( int $order_id ): void {
...
+		$allowed_statuses = [ 'completed', 'failed', 'refunded' ];
+		$subscription     = WCSubscriptions::getSubscriptionForOrder( $order->get_id() );
+
+		// @todo: need to verify it this change is needed.
+		/*
+		if ( $subscription ) {
+			$allowed_statuses[] = 'cancelled';
+			$allowed_statuses[] = 'pending';
+		}
+		*/
```
Comment (pawelkmpt) [inline src/Export/Component.php:214]: @saas786 did you verify? can we remove the commented out code?

**Comment by pawelkmpt** [comment]
> We don't need to self-host Sentry SDK, we're already using https://github.com/stayallive/wp-sentry

@saas786 can you confirm we have necessary dependencies and can drop the line from `composer.json`?

File: src/Export/Component.php (hunk 1 above)
Comment (saas786) [reply src/Export/Component.php:214]: > @saas786 did you verify? can we remove the commented out code?

Did affect the logic, still in progress, most probably will remove.

**Review by pawelkmpt — APPROVED** [review]
@saas786 I approve, because I know it needs to be tested/go live, but please do following things in all future commits.

**1. Make more meaningful commit messages**

It's good you use semantic commit messages prefixes but it'd be better to know what `refactor` is about.

Please take a look at these:
* https://github.com/conversionxl/cxl-playbook-blocks/commits/main
* https://github.com/conversionxl/institute-plugin/commits/master

**2. Squash/amend commits if they're about same thing**

It's difficult to find and explain golden rule, so I'll use an example. These should be squashed:
* 97348c3af32ee1df15c555580edd00b91dd3dfed
* 49eb6b8cf0259599442c5e6939a8b768af9e2ee1

I mentioned recently it's better to have more smaller commits that one huge, but try to find the correct balance.

**3. Fix all coding standards warnings/errors**

It's difficult to read and review the code if every few lines is huge box with error.

**Comment by pawelkmpt** [comment]
@saas786 let me know when it's ready to go live testing

**Comment by saas786** [comment]
> @saas786 let me know when it's ready to be merged and go live testing

Working out some change(s), if its in useable state, will use it in this PR, otherwise, will move it to next PR.

**Comment by saas786** [comment]
> > 1. Fix all coding standards warnings/errors
> 
> still an issue

Please take a look now, resolved 50% plus linting issues highlighted PHPCS test on GH.

**Review by pawelkmpt — CHANGES_REQUESTED** [review]
Use these for DB queries to fix remaining CS issues:

* `WordPress.DB.DirectDatabaseQuery.DirectQuery`
* `WordPress.DB.DirectDatabaseQuery.NoCaching`

> `Use of empty() is disallowed.`

I don't know where does it come from

File: src/Tools/Logger/Component.php [hunk 2]
```diff
@@ -5,17 +5,111 @@
...
+use Monolog\Handler\NullHandler;
 use Monolog\Handler\StreamHandler;
 use Monolog\Logger as MLogger;
+use Sentry\ClientBuilder;
+use Sentry\Serializer\RepresentationSerializer;
+use Sentry\Serializer\Serializer;
+use Sentry\State\Hub;
+use Sentry\State\Scope;
+use WP_Sentry_Php_Tracker;
+use WP_Sentry_Version;
```
[the remark this answers is not in the document]
Comment (pawelkmpt) [reply src/Tools/Logger/Component.php]: @saas786 did you test it?

File: src/Tools/Logger/Component.php [hunk 3]
```diff
@@ -5,17 +5,111 @@
...
+
+	/**
+	 * Singleton pattern
+	 */
+	private static ?MLogger $logger = null;
+
+	/**
+	 * Singleton pattern.
+	 */
+	public static function get_logger(): MLogger {
```
[the remark this answers is not in the document]
Comment (pawelkmpt) [reply src/Tools/Logger/Component.php]: Looking good, but I see one more thing to optimize. Every handler has the following line:

```php
$logger = new MLogger( gmdate( 'Y-m-d' ) . '.log' );
```

**Review by pawelkmpt — CHANGES_REQUESTED** [review]
`phpcs:ignore SlevomatCodingStandard.ControlStructures.DisallowEmpty.DisallowedEmpty` is placed in so many places that it should be:
* dropped from Slevomat,
* ignored with `phpcs:ignoreFile SlevomatCodingStandard.ControlStructures.DisallowEmpty.DisallowedEmpty`

File: src/Schedulers/Actions/Orders.php [hunk 4]
```diff
@@ -41,22 +41,12 @@ public static function init(): void {
 
 		add_action( 'cxl/wc/chartmogul/schedulers/orders/export/failed', static function( array $action_data ): void {
 
-			[ $order_id, $date_time_from, $skip_existing, $resource_id, $data_source, $export_by ] = array_values( $action_data['get_args'] );
+			[ $order_id, , , , ,] = array_values( $action_data['get_args'] );
```
[the remark this answers is not in the document]
Comment (saas786) [reply src/Schedulers/Actions/Orders.php:44]: > wow, what does this do? I don't know this syntax

This article might explain it better: https://www.w3schools.com/php/func_array_list.asp

**Comment by saas786** [comment]
@pawelkmpt please review it now.

**Pull request merged**